apps/user-portal/src/components/account-recovery-code.tsx

```diff
@@ -0,0 +1,230 @@
+            schemas: [
+                "urn:ietf:params:scim:api:messages:2.0:PatchOp"
+            ]
+        };
+
+        const phoneNumbers = {
+            type: "mobile",
+            value: mobile
+        };
+
+        data.Operations[0].value = {
+            phoneNumbers: [
+                {
+                    type: "mobile",
+                    value: editedMobile
+                }
+            ]
+        };
+
+        updateProfileInfo(data)
+            .then((response) => {
+                if (response.status === 200) {
+                    onNotificationFired({
+                        description: t(
+                            "views:securityPage.multiFactor.smsOtp.notification.success.description"
+                        ),
+                        message: t(
+                            "views:securityPage.multiFactor.smsOtp.notification.success.message"
+                        ),
+                        otherProps: {
+                            positive: true
+                        },
+                        visible: true
+                    });
+                    getProfileInfo()
+                        .then((res) => {
+                            setMobileNo(res);
+                        });
+                    setIsEdit(false);
+                } else {
+                    onNotificationFired({
+                        description: t(
+                            "views:userProfile.notification.getProfileInfo.error.description"
+                        ),
+                        message: t(
+                            "views:userProfile.notification.getProfileInfo.error.message"
+                        ),
+                        otherProps: {
+                            negative: true
+                        },
+                        visible: true
+                    });
+                }
+            });
+    };
+
+    /**
+     * The following function handles the change of state of the input fields.
+     * The id of the event target will be used to set the state.
+     * @param event
+     */
+    const handleFieldChange = (event) => {
+        setEditedMobile(event.target.value);
+        event.preventDefault();
+    };
+
+    const setMobileNo = (resp) => {
+        let mobileNumber = "";
+        resp.phoneNumbers.map((mobileNo) => {
+            mobileNumber = mobileNo.value;
+        });
+        setMobile(mobileNumber);
+        setEditedMobile(mobileNumber);
+    };
+
+    const handleEdit = () => {
+        setIsEdit(true);
+    };
+
+    const handleCancel = () => {
+        setIsEdit(false);
+    };
+
+    const showEditView = () => {
+        if (!isEdit) {
+            return (<Grid padded>
+                    <Grid.Row columns={2}>
+                        <Grid.Column width={11} className="first-column">
+                            <List.Content floated="left">
+                                <ThemeIcon
+                                    icon={MFAIcons.sms}
```

Comment:
Let's add a separate icon for recovery code

Reply:
The recovery code section has been removed for the time being.